### ISSUE TITLE
TransmissionVPN loses downloading data on container restart

### ISSUE THREAD

**Issue by hannonq** [GhostWriters/DockSTARTer#771]
**Describe the bug**
Not sure if this a bug or me failing to setup the environment variables and/or volumes correctly.
I just started using DockSTARTer this weekend and I love it, but I have one problem so far:

When I'm downloading a torrent file and I restart the `transmissionvpn` container, the file that was being download starts again with 0% progress:

**To Reproduce**

1. Start transmissionvpn with `sudo ds -c`
2. Add a torrent through the web interface
3. Run `sudo ds -c down` while the download is happening, then run `sudo ds -c` again
4. When I open transmission, the torrent is completely gone. It doesn't even show on the list.

Also:

1. Start transmissionvpn with `sudo ds -c`
2. Add a torrent through the web interface
3. Run `sudo ds -c restart` while the download is happening.
4. When I open transmission, the torrent *is* on the list, but it restarts (automatically) with 0% progress

Problem also happens with `docker stop/start transmissionvpn`, so I'm tempted to think I screwed up somewhere in the configuration.

**Expected behavior**
I expect the torrent download to resume from where it stopped when I stopped/restarted the container

**System (please complete the following information):**

- OS: Ubuntu Server
- Version 18.04

**Additional context**
Relevant Environment Variables (as shown on Portainer):

| Var                           	| Value                                             	|
|-----------------------------	|-----------------------------------------------	|
|  TRANSMISSION_DOWNLOAD_DIR  	|                   /completed                  	|
| TRANSMISSION_INCOMPLETE_DIR 	|                  /incomplete                  	|
|    TRANSMISSION_WATCH_DIR   	|                     /watch                    	|
| TRANSMISSION_HOME           	| /home/hannonq/.config/appdata/transmissionvpn 	|

**Volumes**

Host/volume | Path in container
-- | --
/media/HD4TB1/Downloads/Watch | /watch
/media/HD4TB1/Downloads/Complete | /completed
/home/hannonq/.config/appdata/transmissionvpn | /data
/home/hannonq/.config/appdata/.openvpn | /etc/openvpn/custom
/home/hannonq/.config/appdata/transmissionvpn | /config
/media/HD4TB1/Downloads/Incomplete | /incomplete
/media/HD4TB1/Downloads | /downloads
/home/hannonq/.config/appdata/shared | /shared

**Comment by nemchik** [comment]
Based on what you're posting above I am under the impression that you're using overrides to adjust the environment and volumes. Can you try not using an override? Specifically you want the following variables and volumes (which are set by DS already):


```yaml
    environment:
      - TRANSMISSION_DOWNLOAD_DIR=/downloads/completed
      - TRANSMISSION_INCOMPLETE_DIR=/downloads/incomplete
      - TRANSMISSION_WATCH_DIR=/downloads/watch
```
```yaml
volumes:
      - ${DOCKERCONFDIR}/transmissionvpn:/config
      - ${DOWNLOADSDIR}:/data
      - ${DOWNLOADSDIR}:/downloads
      - ${VPN_OVPNDIR}:/etc/openvpn/custom/
```
And to be honest the /config and /data volumes might not even be needed (but are included by DS).

**Comment by hannonq** [comment]
I have removed the override, restarted the container, made sure that the changes really happened, but I still got the same behavior..

Strangely enough, when I added my override again, I noticed that the downloaded does not go back on the list, like before. BUT, this time when I add the same torrent again, the "verifying local data" phase actually finds the .part and picks up from where I had stopped.

But it's still not ideal, since I have to manually add the torrent again.

**Comment by nemchik** [comment]
The desired result is that the downloads should be saved in the folder you specified as `DOWNLOADSDIR` in DS. You can check this in `~/.docker/compose/.env` right near the top. I'm wondering if you remove everything you had in the override but just put this if it works:

```yaml
    environment:
      - TRANSMISSION_HOME=/config
```

**Comment by hannonq** [comment]
Unfortunately, nothing so far seems to solve the problem.

I have even completely removed Transmission from my DS installation, removed the .config folders related to Transmission... but when I install it again, the same problem occurs.

I found [this issue](https://github.com/haugene/docker-transmission-openvpn/issues/40) on the official repo from the image you use, which links to [this other issue](https://github.com/haugene/docker-transmission-openvpn/issues/23) that seems to have been resolved.

So I don't know what else I can try to do to solve this. Have you checked that this problems does *not* happen with you?

**Comment by nemchik** [comment]
I have not checked this issue personally, but I can give it a shot. TransmissionVPN is the only download client+VPN container in this project that is not made by (or forked from) binhex. All the rest seem to be fairly easy to setup. Maybe give qbittorrentvpn a try? I'll keep this issue open until I'm able to resolve the issues with transmissionvpn though.

**Comment by nemchik** [comment]
Ok, I've got transmissionvpn working and persisting torrent data through recreating the container (fully stopping and removing the container and making a new one).

You can pull these changes by running `ds -u origin/transmissionvpn`

You should not need an override at all with these changes. I'll get them merged soon.

**Comment by hannonq** [comment]
I've pulled the changes (now from master, since I see you already merged it) and the download still doesn't restart automatically once the container is created :/

The data is indeed persisted, because when I manually add the torrent again (via magnetic link), it does pick up from where it had stopped. But I still have to add the torrent again, manually.

Did you also test it for this case?

In the meantime, I'll give qbittorrentvpn a try.

**Comment by nemchik** [comment]
This is what I tested:

I enabled the app via DS and setup the required variables and ovpn file so I could access the Transmission web interface (via reverse proxy in my case).
I added a torrent
I paused the torrent
I stopped the container
I removed the container
I had DS recreate the container
I accessed the Transmission web interface again
The torrent was still in the interface paused

I did not test without pausing, but I did not have to re-add the torrent.

Again I did not need to use any overrides for this.
